File: EDR_NDR/backend/app/schemas/event.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import EventType
from app.schemas.agent import AgentRef


class EventOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    timestamp: datetime
    agent_id: str
    agent: AgentRef | None = None
    event_type: EventType
    process_name: str | None
    command_line: str | None
    parent_process: str | None
    user: str | None
    src_ip: str | None
    dest_ip: str | None
    dest_port: int | None
    protocol: str | None
    file_hash: str | None
    file_path: str | None
    raw: str
    summary: str = ""


class EventContextResponse(BaseModel):
    anchor_event: EventOut
    before: list[EventOut]
    after: list[EventOut]

File: EDR_NDR/frontend/src/types/event.ts
import type { AgentRef } from './agent'
import type { EventTypeValue } from '../lib/labels'

export interface EventOut {
  id: string
  timestamp: string
  agent_id: string
  agent: AgentRef | null
  event_type: EventTypeValue
  process_name: string | null
  command_line: string | null
  parent_process: string | null
  user: string | null
  src_ip: string | null
  dest_ip: string | null
  dest_port: number | null
  protocol: string | null
  file_hash: string | null
  file_path: string | null
  raw: string
  summary: string
}

export interface EventContextResponse {
  anchor_event: EventOut
  before: EventOut[]
  after: EventOut[]
}

export interface EventSearchParams {
  start_time: string
  end_time: string
  agent_id?: string
  event_type?: EventTypeValue[]
  keyword?: string
  page?: number
  page_size?: number
}
